docs/approach-to-analytics.md

```diff
@@ -0,0 +1,135 @@
+# Approach to Analytics
+
+Content Publisher uses [Google Analytics](https://analytics.google.com/analytics/web/#/report-home/a124922226w186404040p183397753) (GA) to store events that describe a user action, such as when a user clicks the "[Create new document](https://github.com/alphagov/content-publisher/blob/4b088834621787009528afecb4c3fe876b9cf577/app/views/documents/index.html.erb#L4)" button. Rather than dispatching events to GA directly, we use [Google Tag Manager](https://tagmanager.google.com/#/container/accounts/3880990677/containers/9704122) (GTM) to reduce the need for custom analytics code in the app. In order to keep our analytics robust, we use special 'data-gtm' attributes to consistently identify elements on a page, as other attributes, like CSS classes, will change over time.
+
+> Data submitted by the user should not be tracked. Such data is already stored in the relational database for the app. Developers can provide support on analytics work involving this kind of data.
+
+> Details on how and why our analytics works this way can be found in individual PR and GTM version descriptions, which are all documented on the [Trello card](https://trello.com/c/B9rVHJTh/876-audit-and-fix-gtm-ga-issues) that motivated the work we did to get to this point.
+
+## Tracking at a High Level
+
+We track most events using a standard set of `data-gtm` attributes.
+
+   - `data-gtm` describes the intent of the element (*required*)
+   - `data-gtm-value` indicates the value in a choice (optional)
+   - `data-gtm-click-tracking=true` is used to track custom elements
+   - `data-gtm-visibility-tracking=true` tracks when an element appears
+
+GTM is then setup to use these standard attributes in the events it sends to GA.
+
+   - **Category** = `data-gtm`
+   - **Action** = Click, Visible, Check, Uncheck
+   - **Label** = `data-gtm-value`
+
+We also send [additional fields](https://github.com/alphagov/content-publisher/blob/4b088834621787009528afecb4c3fe876b9cf577/app/views/layouts/application.html.erb#L9) to GA that describe the page itself.
+
+   - **User Organisation** e.g. `government-digital-service`
+   - **User ID** e.g. `c78492f0-0356-0136-a017-02ada856ca8e`
+    - This is also used to [enable a bunch of GA User-ID features](https://support.google.com/analytics/answer/3123662?hl=en)
+   - **Page Name** e.g. `documents-show`
+    - This is also used to [automatically specify content groupings](https://support.google.com/analytics/answer/2853423?hl=en)
+
+
+## Examples for Elements
+
+Currently it's not clear what data we need to collect. Until the app reaches maturity, **we've decided to track all user actions**. The following is a list of what this means in terms of HTML Elements.
+
+### Links
+
+```
+<a ... data-gtm="new-document" ...>Create new document</a>
```

Comment:
So it sounds like you think there's some ambiguity in the use of this attribute. I thought this would be covered by the definition that "`data-gtm` describes the intent of the element" - different elements ought to have different intents, so shouldn't end up with the same values for this attribute. Do you think this needs more emphasising?

Reply:
Yeah, it might be worth expanding on the data-gtm to indicate it should be unique. I think intent doesn't quite cover it. For example we have two elements on the page to "Raise a support request", both have same intent but have different data-gtm values.

Reply:
Good example. I realise it also leads to the ambiguity for radio buttons later on, so I've extended this section to be a bit more flexible and to give examples.